src/redux/missions/missionSlice.js

```diff
@@ -0,0 +1,58 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+export const getMission = createAsyncThunk('missions/getMission', async () => {
+  try {
+    const res = await axios.get('https://api.spacexdata.com/v3/missions');
+    return res.data;
+  } catch (error) {
+    throw Error('Error fetching missions');
+  }
+});
+
+const initialState = {
+  missions: [],
+  isLoading: false,
+  isError: null,
+};
+
+const missionSlice = createSlice({
+  name: 'Mission',
+  initialState,
+  reducers: {
+    reserve: (state, action) => {
+      state.missions = state.missions.map((mission) => {
+        if (mission.mission_id === action.payload) {
+          return { ...mission, reserved: true };
+        }
+        return mission;
+      });
+    },
+    cancel: (state, action) => {
+      state.missions = state.missions.map((mission) => {
+        if (mission.mission_id === action.payload) {
+          return { ...mission, reserved: false };
+        }
+        return mission;
+      });
+    },
+  },
```

Comment:
- [ ] Please 🙏🏽 add tests for your functions reserve and cancel, it is a project requirement for this capstone 👇🏽 

> NOTE: Make sure to add tests for all your components and functions.

Reply:
![Screenshot 2023-08-31 005549](https://github.com/SnakyMz/space-hub/assets/115364362/4f6cfd0b-abd6-42fd-9663-9931a8ff6850)
Changes as suggested